torch/functional.py

```diff
@@ -910,5 +912,112 @@ def lu(A, pivot=True, get_infos=False, out=None):
         return result[0], result[1]  # A_LU, pivots
 
 
+def clamp(input, min=None, max=None, out=None):
+    r"""
```

Comment:
This doc string looks pretty good, but do you need more detail about clamp's behavior on quantized and complex values? In particular, does clamping a complex value work like clamping it in Numpy? If not we should be specific about the behavior so the user knows what to expect. cc @anjali411, who's looking at complex-behavior in PyTorch.

Reply:
I have merely copied the existing docstring and added examples with tensor arguments, the description is more or less the same. 

With regards to quantized tensors, I think everything is already explained very well in the main document about [quantization](https://pytorch.org/docs/stable/quantization.html). In particular this paragraph:

>PyTorch also supports quantization aware training, which models quantization errors in both the forward and backward passes using fake-quantization modules. Note that the entire computation is carried out in floating point. At the end of quantization aware training, PyTorch provides conversion functions to convert the trained model into lower precision.

Would it then be useful to repeat it here too? I would argue that if all quantized operations behave in the same way, then it is not necessarily important to repeat it in each operation's docstring. I am happy to add it if you think it would be helpful.

With regards to complex values, since it is now a disabled feature following the recent discussions, I am not adding it to the doc.

Reply:
That sounds good. What's the reason for adding clamp here, as opposed to keeping its documentation where it was?

Reply:
I've added an override of the default `torch.clamp` here to dispatch to `at::clamp_with_tensors` or `at::clamp` depending on the argument types. The build required me to move the docstring here too as otherwise I got some complaint. It is also equivalent to the other functions that are overridden/defined in `functional.py`, where their docstrings are defined within the function and not in a separate file.

Reply:
Interesting. That's cool.